examples/flash-sale/pages/collections/[handle].js

```diff
@@ -0,0 +1,233 @@
+import { useState } from 'react';
+import { useRouter } from 'next/router';
+import { nacelleClient } from 'services';
+import ProductCard from 'components/ProductCard';
+import styles from 'styles/Collection.module.css';
+import Countdown, { formatTimeDelta, zeroPad } from 'react-countdown';
+
+function Collection(props) {
+  const router = useRouter();
+  const { collection, flashSale } = props;
+  const [products, setProducts] = useState(props.products);
+  const [canFetch, setCanFetch] = useState(props.canFetch);
+  const [isFetching, setIsFetching] = useState(false);
+
+  const activeProducts = canFetch
+    ? products?.slice(0, products.length - 1)
+    : products;
+
+  let flashSaleText = false;
+  const flashSaleRenderer = (timeDelta) => {
+    const formattedDate = formatTimeDelta(timeDelta, {
+      daysInHours: true
+    });
+    const { hours, minutes, seconds, completed } = formattedDate;
+    if (completed) {
+      return <></>;
+    } else {
+      return (
+        <p suppressHydrationWarning>
```

Comment:
To Alex's point about rendering only on the client, one way we could accomplish that would be to leverage Webpack's `process.browser`:

```suggestion
    if (completed || !process.browser) {
      return <></>;
    } else {
      return (
        <p>
```

IMHO that'd be preferable to suppressing hydration warnings.

Reply:
> To Alex's point about rendering only on the client, one way we could accomplish that would be to leverage Webpack's `process.browser`:
> 
> IMHO that'd be preferable to suppressing hydration warnings.

Should we use `typeof window !== undefined`?  Seems like the `process.browser` is being deprecated:
https://github.com/vercel/next.js/issues/5354#issuecomment-520305040

Reply:
Yes, good catch @alexcasche! I've updated the suggestion accordingly.

Reply:
While this does work, it leaves me with a new error in the console.

<img width="647" alt="Screen Shot 2022-03-09 at 9 12 24 AM" src="https://user-images.githubusercontent.com/1411409/157494795-46909043-78eb-4d76-910d-956ff255605f.png">

Reply:
@alexcasche @NWRichmond I assume what's happening here is that it renders on the server with no elements, but on the client it has the `<p>` and `<strong>` tag from the countdown timer markup.

The use of `suppressHydrationWarning` was to get around the timer running and being different between server and client. I've tried a different approach, having the timer not automatically start, and start it manually, but was running into the same warning error of the timer being different.

Reply:
Got a new solution @jasonredick @NWRichmond .  Kind of annoying but seems like the Next.js way.

```
// Step 1:  Add new piece of state
const [showCountdown, setShowCountdown] = useState();
```

```
// Step 2:  Toggle to true in useEffect
useEffect(() => {
    setShowCountdown(true)
  }, [])
```

```
// Step 3:  Use this variable to conditionally render content
if (completed || !showCountdown) {
      return <></>;
    } else {
      return (
        <p>
          <strong>Sale Ends</strong> {zeroPad(hours)}:{zeroPad(minutes)}:
          {zeroPad(seconds)}
        </p>
      );
    }
```

Reply:
@alexcasche That looks like it worked, thank you!